test_contextual_lenses/tape_gfp.py

```diff
@@ -0,0 +1,673 @@
+"""Contextual lenses on TAPE fluoresence dataset."""
+
+
+import jax.numpy as jnp
+
+import numpy as np
+
+import tensorflow as tf
+
+import numpy as np
+
+import pandas as pd
+
+import matplotlib.pyplot as plt
+
+import scipy.stats
+
+import sklearn.linear_model
+import sklearn.metrics
+
+import seaborn as sns
+
+from contextual_lenses import mean_pool, max_pool, \
+linear_max_pool, linear_mean_pool, gated_conv 
+
+from train_utils import create_optimizer, train_step, \
+train, create_representation_model
+
+from encoders import one_hot_encoder, cnn_one_hot_encoder, \
+one_hot_pos_emb_encoder, cnn_one_hot_pos_emb_encoder
+
+from tape.tape.datasets import LMDBDataset
+
+
+"""Data processing."""
+def dataset_to_df(in_name):
+  dataset = LMDBDataset(in_name)
+  df = pd.DataFrame(list(dataset)[:])
+  df['log_fluorescence'] = df.log_fluorescence.apply(lambda x: x[0])
+  df['binary_log_f'] = [int(b) for b in df['log_fluorescence'] > 2.5]
+  return df
+
+
+"""Padding."""
+SEQ_LEN = 237
+def pad_seq(seq, pad_char='-'):
+  
+  padded_seq = seq + pad_char*(SEQ_LEN-len(seq))
+  padding_mask = []
+  for _ in range(len(seq)):
```

Comment:
you can avoid these for loops. I'd just do padding_mask = np.int32(padded_seq == pad_char).